### DISCUSSION TITLE
Compatible insert with auto_id and non auto_id

### DISCUSSION THREAD

**Discussion by fightingyuman** [milvus-io/milvus#15672] [Q&A and General discussion] [answered]
Is there any way to insert data by pymilvus 2.0.0 which support the data **with** primary field and **without** primary field, just like mysql auto increment primary key.

eg.
```python
# used auto_id for primary field.
field1 = FieldSchema(name="id", dtype=DataType.INT64, descrition="primary field", is_primary=True, auto_id=True)
field2 = FieldSchema(name="faceFeature", dtype=DataType.FLOAT_VECTOR, descrition="float vector", dim=2, is_primary=False)
schema = CollectionSchema(fields=[field1, field2], description="collection")
collection = Collection(name=self.collection_name, schema=schema)

# I want to support both the following insert:
# 1. the server auto generate primary field:
vector = [0.23, 0.52]
collection.insert([vector])
# 2. import with user specified primary field
my_id = 1
vector = [0.23, 0.52]
collection.insert([my_id], [vector])
```

I want to use it here:
We've test that Milvus may hangs insert or search after host lost power, until now milvus don't support backup and restore, we do backup ourself. We back up auto primary key and float vector to some other place, if Milvus hangs, we can restore Milvus data from backup data. When restoring, we wan't to specified primary field. After restore, we wan't use auto primary field.

**Comment by yhmo** [comment, marked as answer]
- If the **auto_id = true**:
```
field1 = FieldSchema(name="id", dtype=DataType.INT64, descrition="primary field", is_primary=True, auto_id=True)
field2 = ...
```

Call insert() in this way:
```
entities = [
  [vector1, vector2, vector3 ...]
]
collection.insert(entities)
```


- If the **auto_id = false**:
```
field1 = FieldSchema(name="id", dtype=DataType.INT64, descrition="primary field", is_primary=True, auto_id=False)
field2 = ...
```

Call insert() in this way:
```
entities = [
  [id1, id2, id3 ...],
  [vector1, vector2, vector3 ...]
]
collection.insert(entities)
```

**Reply by fightingyuman** [reply]
Thanks, the insert works. Finally, we set `auto_id =  False` and generate Primary key ourself.
I want **both** insert can be finished when **auto_id = True**. I following pymilvus, it can't pass the data validate check when entities contains id and `auto_id = True`. Just like what mysql auto_increment primary key does.

**Reply by yhmo** [reply]
About "restore", do you mean to create a new collection to import the backup data?
